src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java

```diff
@@ -380,4 +380,52 @@ void expandBracketsWithTestCasesFromRegExpBasedFileFinder() {
                 "Eric von Hippel and Georg von Krogh have published Open Source Software and the \"Private Collective\" Innovation Model: Issues for Organization Science in Organization Science.",
                 BracketedPattern.expandBrackets("[author] have published [title] in [journal].", ',', entry, database));
     }
+
+    @Test
+    void expandBracketsWithFallbackToString() {
+        BibEntry bibEntry = new BibEntry()
+                .withField(StandardField.TITLE, "Title");
+
+        assertEquals("Title_eprint", BracketedPattern.expandBrackets("[title]_[EPRINT:(eprint)]", null, bibEntry, null));
```

Comment:
Thank you so much for test cases.

These test cases show that `null` is provided as parameter. This should not be done. Please introduce other methods not requiring to pass `null`. Maybe, the parameters can be removed at these methods and `null` passed to the four-paramters-method?